File: Runtime/Plugins/Android/VarioqubProxy.java
package com.yandex.varioqub.plugin.unity;

import android.app.Activity;
import android.util.Log;

import androidx.annotation.NonNull;

import com.yandex.varioqub.analyticadapter.VarioqubConfigAdapter;
import com.yandex.varioqub.appmetricaadapter.AppMetricaAdapter;
import com.yandex.varioqub.config.Varioqub;
import com.unity3d.player.UnityPlayer;

import org.json.JSONException;

import java.util.Set;

public class VarioqubProxy {

    public static void initVarioqubWithAppMetricaAdapter(@NonNull String settingsJson) {
        init(settingsJson, new AppMetricaAdapter(getActivity()));
    }

    public static void fetchConfig(@NonNull OnFetchCompleteListenerProxy proxy) {
        Varioqub.fetchConfig(new OnFetchCompleteListenerProxy.Delegate(proxy));
    }

    public static void activateConfig() {
        Varioqub.activateConfig(null);
    }

    public static void setDefaults(@NonNull String defaultsJsonValue) {
        try {
            Varioqub.setDefaults(JsonUtils.convertJsonStringToMap(defaultsJsonValue));
        } catch (JSONException e) {
            Log.e("VarioqubUnity", "Something went wrong while parsing defaults map", e);
        }
    }

    @NonNull
    public static String getString(@NonNull String key, @NonNull String defaultValue) {
        return Varioqub.getString(key, defaultValue);
    }

    public static double getDouble(@NonNull String key, double defaultValue) {
        return Varioqub.getDouble(key, defaultValue);
    }

    public static long getLong(@NonNull String key, long defaultValue) {
        return Varioqub.getLong(key, defaultValue);
    }

    public static boolean getBoolean(@NonNull String key, boolean defaultValue) {
        return Varioqub.getBoolean(key, defaultValue);
    }

    @NonNull
    public static String getId() {
        return Varioqub.getId();
    }

    public static void putClientFeature(@NonNull String key, @NonNull String value) {
        Varioqub.putClientFeature(key, value);
    }

    public static void clearClientFeatures() {
        Varioqub.clearClientFeatures();
    }

    @NonNull
    public static String[] getAllKeys() {
        Set<String> keys = Varioqub.getAllKeys();
        return keys.toArray(new String[0]);
    }

    private static void init(String settings, VarioqubConfigAdapter adapter) {
        try {
            Varioqub.init(JsonUtils.convertSettingsFromJson(settings), adapter, getActivity());
        } catch (JSONException e) {
            Log.e("VarioqubUnity", "Something went wrong while parsing Varioqub Settings", e);
        }
    }

    private static Activity getActivity() {
        return UnityPlayer.currentActivity;
    }
}

File: Runtime/Native/Android/Proxy/VarioqubProxy.cs
#if UNITY_ANDROID
using UnityEngine;

namespace Com.Yandex.Varioqub.Native.Android.Proxy {
    internal static class VarioqubProxy {
        private static readonly AndroidJavaClass NativeClass = new AndroidJavaClass("com.yandex.varioqub.plugin.unity.VarioqubProxy");
        
        public static void InitVarioqubWithAppMetricaAdapter(string settings) {
            NativeClass.CallStatic("initVarioqubWithAppMetricaAdapter", settings);
        }

        public static void FetchConfig(OnFetchCompleteListenerProxy fetchCompleteListenerProxy) {
            NativeClass.CallStatic("fetchConfig", fetchCompleteListenerProxy);
        }

        public static void ActivateConfig() {
            NativeClass.CallStatic("activateConfig");
        }

        public static bool GetBoolean(string key, bool defaultValue) {
            return NativeClass.CallStatic<bool>("getBoolean", key, defaultValue);
        }

        public static double GetDouble(string key, double defaultValue) {
            return NativeClass.CallStatic<double>("getDouble", key, defaultValue);
        }

        public static long GetLong(string key, long defaultValue) {
            return NativeClass.CallStatic<long>("getLong", key, defaultValue);
        }

        public static string GetString(string key, string defaultValue) {
            return NativeClass.CallStatic<string>("getString", key, defaultValue);
        }

        public static string GetId() {
            return NativeClass.CallStatic<string>("getId");
        }

        public static void SetDefaults(string defaultsJsonValue) {
            NativeClass.CallStatic("setDefaults", defaultsJsonValue);
        }

        public static void PutClientFeature(string key, string value) {
            NativeClass.CallStatic("putClientFeature", key, value);
        }

        public static void ClearClientFeatures() {
            NativeClass.CallStatic("clearClientFeatures");
        }

        public static string[] GetAllKeys() {
            return NativeClass.CallStatic<string[]>("getAllKeys");
        }
    }
}
#endif
